Python/sysmodule.c

```diff
@@ -1291,19 +1291,22 @@ static PyStructSequence_Desc asyncgen_hooks_desc = {
     2
 };
 
-static PyObject *
-sys_set_asyncgen_hooks(PyObject *self, PyObject *args, PyObject *kw)
-{
-    static char *keywords[] = {"firstiter", "finalizer", NULL};
-    PyObject *firstiter = NULL;
-    PyObject *finalizer = NULL;
+/*[clinic input]
+sys.set_asyncgen_hooks
 
-    if (!PyArg_ParseTupleAndKeywords(
-            args, kw, "|OO", keywords,
-            &firstiter, &finalizer)) {
-        return NULL;
-    }
+    firstiter: object = NULL
+    finalizer: object = NULL
+
+set_asyncgen_hooks(* [, firstiter] [, finalizer])
```

Comment:
Maybe the doc should be modified to use this signature: https://docs.python.org/dev/library/sys.html#sys.set_asyncgen_hooks

Or this docstring should use the doc signature: `sys.set_asyncgen_hooks(firstiter, finalizer)`.

Reply:
The docs also read:

> Accepts two optional keyword arguments [...]

IMO, the signature in the docs should be updated to reflect the actual implementation.

Reply:
Can you please rebase your PR on top of it? There are now merge conflicts.